sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/continuous/ContinuousExecution.scala

```diff
@@ -390,6 +398,41 @@ class ContinuousExecution(
     }
   }
 
+  /**
+   * Stores error and stops the query execution thread to terminate the query in new thread.
+   */
+  def stopInNewThread(error: Throwable): Unit = {
+    failureLock.synchronized {
+      failure match {
+        case None =>
+          logError(s"Query $prettyIdString received exception $error")
+          failure = Some(error)
+          stopInNewThread()
+        case _ =>
+          // Stop already initiated
+      }
+    }
+  }
+
+  /**
+   * Stops the query execution thread to terminate the query in new thread.
+   */
+  private def stopInNewThread(): Unit = {
+    new Thread("stop-continuous-execution") {
```

Comment:
This is a pretty common pattern, I'm surprised there isn't a helper method for it.

`ThreadUtils.runInNewThread` would be perfect except it joins the new thread (!!!!), making it sort of useless...

Reply:
Before I've created this PR checked this function but not used because of the join. I've considered to modify it by adding a `shouldJoin` parameter with default `true` but always afraid of changing such core parts. If you think it worth it can be done though.

Reply:
Hmmm, seems like the result has to be abandoned as well so a new function is required if we want this to be common.

Reply:
Proper thing would be for `runInNewThread` to return a `Future` and let the caller decide what to do. But ok to punt on that change.

Reply:
I've taken a deeper look at this part and having fear to break something. To be on the safe side I would not change such core functionality even if the change is relatively small.